src/lando/main/models/repo.py

```diff
@@ -100,6 +100,33 @@ def path(self) -> str:
     # string as falsey.
     push_target = models.CharField(blank=True, default="")
 
+    # If this repo was migrated from another repo, link to it here.
+    # If this value is set, then revisions targeting legacy_source will
+    # have their target repo switched to `self` upon triggering a landing.
+
+    # For example, if `self` is `NewGitRepo` and `self.legacy_source` is `OldHgRepo`
+    # then:
+    # - `NewGitRepo.is_legacy` is False
+    # - `OldHgRepo.is_legacy` is True
+    # - `NewGitRepo.target_repo` is not defined
+    # - `OldHgRepo.target_repo` is set to `NewGitRepo`
+    # A revision that was created against `OldHgRepo` will land in `NewGitRepo`.
```

Comment:
Nit: Would having legacy repos pointing to legacy repos be a risk to protect against?